### ISSUE TITLE
Getting further array-like operations working for astropy.modeling model sets

### ISSUE THREAD

**Issue by emsellem** [astropy/astropy#4100]
Model sets in astropy are a great addition, which allows to have a set of models (defined by the same function) within one astropy structure. Hence we can do: ModelSet = MyModel(par1=[a1,a2,a3],par2=[b1,b2,b3]). However, it is not yet (easily) possible to then slice the ModelSet to:
- do a simple slicing: ModelSubSet = ModelSet[0,2] or ModelSubSet=ModelSet[-1]
- or easily add a new one NewModelset = ModelSet  + MyModel(par1=[a4,a5],par2=[b4,b5])

Having such options would be a fantastic addition I think. This would make a model set a very flexible tool to manipulate N times the same MyModel and associated subsets.

**Comment by embray** [comment]
The syntax `NewModelset = ModelSet + MyModel(par1=[a4,a5],par2=[b4,b5])` isn't going to work in that way--addition of numpy arrays adds those arrays element-wise (it doesn't concatenate them).  Addition of models with `+` is already defined as implementing a compound model, which is somewhat more analogous with element-wise addition of models (granted, addition of the results of models and not, say, the parameters, which is generally less useful).

More likely some kind of concatenate method will be needed as I suggested in #4096 which is relevant to this.

**Comment by emsellem** [comment]
Yes you're right of course. This is the only way also to insure that you are adding sets of parameters to the same model. So something like ModelSet.concatenate or ModelSet.expand would probably be more appropriate. I think "expand" would make more sense (or something like this) as what you are after is not to add two ModelSets (for which you would need then to check the consistency) but to expand on one ModelSet just adding a new series of similar models.

With the remove and slicing this would mean that the following commands:

>>> newSet = ModelSet[list_of_indices]
>>> newSet = ModelSet[:3]

then
>>> ModelSet.delete(list_of_indices)

and
>>> ModelSet.expand(par1=[a4,a5], par2=[b4,b5])

would make sense?
